\PassOptionsToPackage{unicode=true}{hyperref} % options for packages loaded elsewhere
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[12pt,]{article}
\usepackage[]{mathpazo}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provides euro and other symbols
\else % if luatex or xelatex
  \usepackage{unicode-math}
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage[]{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\usepackage{hyperref}
\hypersetup{
            pdftitle={Do androids dream of electric sorhgum?},
            pdfauthor={Ryan P Bartelme; David S LeBauer; Tyson Lee Swetnam; Michael Behrish; Emily Cain; Ishita Debnath; Pankaj Jaiswal; Ab Mosca; Monica Munoz-Torres; Kent Shefchek; P. Bryan Heidorn; Remco Chang; Arun Ross; Anne E Thessen},
            pdfkeywords={machine learning, genomics, phenotype},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage[margin=1in]{geometry}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

% set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother

\usepackage[sort&compress, numbers]{natbib}
\usepackage{etoolbox}
\makeatletter
\providecommand{\subtitle}[1]{% add subtitle to \maketitle
  \apptocmd{\@title}{\par {\large #1 \par}}{}{}
}
\makeatother
\usepackage[]{natbib}
\bibliographystyle{plantphenomics}

\title{Do androids dream of electric sorhgum?\thanks{Replication files are available on the corresponding author's Github
account}}
\providecommand{\subtitle}[1]{}
\subtitle{Predicting Phenotype from Multi-Scale Genomic and Environment Data using
Neural Networks and Knowledge Graphs}
\author{Ryan P Bartelme\footnote{Corresponding Author
  \href{mailto:rbartelme@arizona.edu}{\nolinkurl{rbartelme@arizona.edu}},
  University of Arizona} \and David S LeBauer\footnote{University of Arizona} \and Tyson Lee Swetnam\footnote{University of Arizona} \and Michael Behrish \and Emily Cain \and Ishita Debnath \and Pankaj Jaiswal \and Ab Mosca \and Monica Munoz-Torres \and Kent Shefchek \and P. Bryan Heidorn \and Remco Chang \and Arun Ross \and Anne E Thessen\footnote{Oregon State University}}
\date{March 25, 2020}

\begin{document}
\maketitle

\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

During this period of unprecedented anthropogenic climate change,
understanding genomic response to environmental variation and the
influences on organismal phenotypes is critical. Predicting these
responses is invaluable to maintain the innumerable services natural
ecosystems provide. Multi-scale effects over time and space combined
with the non-linearity of natural systems
\citep{lorenz1963deterministic, ruel1999jensen, west2009general}
obscures the signal of biological processes, their interactions with the
environment, and resulting observable phenotypes.

Existing models for predicting phenotypes from genetic and environmental
data focus on single species or single ecosystems. As we enter an era of
Big Data in ecological research \citep{balch2020neon}, ecologists are
shifting away from relatively small data sets toward national and global
efforts such as the National Ecological Observatory Network (NEON)
\citep{keller2008continental} and airborne and space-based Earth
Observation Systems (EOS). Both societal need and technical capabilities
are moving toward addressing larger scale questions that require
integration of multi-modal data and non-linear predictive models to
understand the interactions between an organism's genetic potential and
its environment and how those interactions result in observable
phenotypes. Ontologies and knowledge graphs are now available for data
integration at scale (e.g., Mungall et al.~2010, Stucky et al.~2018),
enabling an increasing scope beyond single taxon or single ecosystem
models.

\hypertarget{challenges-dataset-interoperability}{%
\section{Challenges Dataset
Interoperability}\label{challenges-dataset-interoperability}}

Incorporating genomic data into phenomics is challenging. Many recent
studies have used only environmental features and machine learning to
predict phenotypes of lilacs, honeysuckle, rice, and wheat
\citep{ALDERMAN20171, nissanka2015calibration, mehdipoor2019geocomputational}.
There are a number of methods linking genomic data to environments or
traits. For example, genome wide association studies (GWAS) enable
researchers to examine the influence of single nucleotide polymorphisms
(SNP) on phenotypes in both natural and controlled settings
\citep{beyer2019loci, schlappi2017assessment, spindel2016genome}. GWAS
often provides generalized and mixed linear model associations between
SNPs and environmental variables, roughly analogous to Genes +
Environment = Phenotype (G+E=P). There are limitations in the
assumptions made by existing methodologies, such as GWAS, that directly
attribute plant phenotypes to environmental variables. These methods do
not explicitly incorporate biological and molecular interactions, such
as post-translational modification of macromolecules
\citep{running2014role}, the importance of plant-microbe interactions
\citep{oyserman2019extracting}, or endogenous siRNA
\citep{katiyar2006pathogen}. However, a machine learning approach allows
for these biological phenomena to be accounted for as latent variables
while probing the interactions of genomes, environments, and phenotypes
in a multidimensional manner.

Conventional observations and statistical models are shifting toward
remotely sensed observation and trait collection, which rely on machine
learning (ML) and computer vision for measurements. For example,
Bayesian Belief Networks \citep{cooper1990computational} may be
implemented to associate environmental variables with traits, and
Generative Adversarial Networks \citep{radford2015unsupervised} for
classifying plant phenotype imaging. Rather than simply generating large
quantities of machine readable data \citep{hampton2013big} and
implementing ML methods ad-hoc \citep{pichler2020machine}, ecologists
are now grappling with how to interpret the massive quantities of
unstructured data that are available at scale.\\
ML predictions often rely on complex, ``black box'' methodologies to
assess explanatory variables. Here we introduce the GenoPhenoEnvo
project, an effort to predict phenotype from genetic and environmental
data, while developing novel representations of the ML ``black-box''
internals.

\hypertarget{future-directions}{%
\section{Future Directions}\label{future-directions}}

Our project has the long-term goal of developing predictive analytics
based on an organisms' genetic code and its associated phenotypic
response to environmental change. To design an initial analytical
framework and workflow, we will first use phenomic, genomic, and
environmental data about sorghum (\emph{Sorghum bicolor}). These data
are available through the TERRA-REF (Transportation Energy Resources
from Renewable Agriculture Phenotyping Reference Platform) project
\citep{lebauer2020data, burnette2018terra}. After training the ML model
on the highly controlled and thorough TERRA-REF data set, we aim to test
and further develop the model with data from less controlled and lower
resolution environments using data from sources such as NEON and the
National Phenology Network.

The first challenge is to prepare these data for use as input into ML
models. In addition to empirical data, ontologically-supported knowledge
graphs can be used to inform the ML \citep{mungall2017monarch}.
Knowledge graphs (KG) are directed acyclic graphs that represent
knowledge in a computational format and are an integral part of Google's
Answer Box and IBM's Watson. KGs can help constrain and prioritize
results, provide quality control, fill in data gaps with inferencing,
and integrate heterogeneous data. In this project, KGs provide an easier
way to integrate data from established plant phenome databases such as
Planteome \citep{cooper2018planteome}, Gramene
\citep{jaiswal_gramene_2011}, and TAIR \citep{poole_tair_2007}. The true
power of ontologies and KGs is a formal logical structure, enabling
inferential and similarity analyses
\citep{mungall2017monarch, washington2009linking}. In particular, it is
this latter feature that will enable data set interoperability. As we
begin to make predictions in more complicated systems with multiple
species and heterogeneous data, the knowledge graphs will be critical
for managing phenotype data.

The GenoPhenoEnvo (GPE) project aims to predict phenotype with genomic
and environmental data using a multimodal approach to training ML
models. We are actively developing a visualization tool to increase
understanding of why the model gave a particular result. In this way,
the GPE project will work toward phenotype predictions and an increased
understanding in the biological and molecular processes that translate
genotype to phenotype. In addition to increased awareness of molecular
effects, the ML models could enable specific ecological hypothesis
testing or predicting long-term speciation events driven by
environmental factors. Ultimately, we believe this combination of
methods will generate a new scientific sub-discipline, one we have
called ``\emph{Computational Ecogenomics.}''

\hypertarget{author-contributions}{%
\section{Author contributions}\label{author-contributions}}

The ordering of authors following RB is alphabetical.

RPB: Conceptualization, Writing-original draft DSL: Conceptualization,
Funding Acquisition, Writing-review\&editing TLS: Conceptualization,
Funding Acquisition, Writing-review\&editing MB: Conceptualization EC:
Conceptualization ID: Conceptualization PJ: Conceptualization, Funding
Acquisition AM: Conceptualization MMT: Conceptualization, Funding
Acquisition, Project Administration KS: Conceptualization PBH:
Conceptualization, Funding Acquisition RC: Conceptualization, Funding
Acquisition AR: Conceptualization, Funding Acquisition AET:
Conceptualization, Funding Acquisition, Supervision,
Writing-review\&editing

\renewcommand\refname{References}
\bibliography{references.bib}

\end{document}
